dvc/fs/webdav.py

```diff
@@ -35,9 +35,11 @@ def __init__(self, **config):
 
     @classmethod
     def _strip_protocol(cls, path: str) -> str:
```

Comment:
I am still confused by the whole situation with `_strip_protocol()`. The `_strip_protocol` has two use cases right now:

1. In [`fsspec.spec.AbstractFileSystem`](https://github.com/fsspec/filesystem_spec/blob/8f62e479ddd45285b3e7126e9313151f1bebd3b9/fsspec/spec.py#L160), this is used internally to strip protocols, i.e. `<scheme>://path` to just `path`.
2. In DVC, to get `fs_path` which DVC uses as a root path/fs_path. https://github.com/iterative/dvc/blob/81eee059950d6a154327e3b15912a84d4b93e178/dvc/fs/__init__.py#L117

From [the docs on `_strip_protocol`[(https://github.com/fsspec/filesystem_spec/blob/8f62e479ddd45285b3e7126e9313151f1bebd3b9/fsspec/spec.py#L161-L164):

> Turn path from fully-qualified to file-system-specific
> May require FS-specific handling, e.g., for relative paths or links.
```

It may seem like 1 and 2 are the same thing (as per docs), but they really aren't.
Regarding the above 2nd point with DVC's usecase, we can just set `fs_path`/`root_path` to be root (with `""` aka `root_marker`).

Given our plans to make it more fsspec-compatible, I don't want to ignore the first use case of `_strip_protocol()` (and, want to avoid wrongfully triggering `_strip_protocol` for other use-case). The `_strip_protocol` also being a class method does not have context about `base_url` to handle this. 

To me, given how we are using this, what we really need is `get_root_path` that defaults to `_strip_protocol` for other filesystems, and for prefixed filesystem, it returns root_marker.

Alternatively, we could add support for PrefixedFileSystem in `odb` itself as proposed before, by setting `fs_path` to a root_marker.

Reply:
I agree that we could use DirFileSystem, but it doesn't seem to solve the problem with the filesystem itself. It looks like we are struggling becuase of the way our config expects webdav remotes to be configured. We currently expect:

```
url = webdavs://example.com...
```

which creates a problem of distinguishing path from base url, but if we would to configure it like

```
url = webdavs://path/to/
base_url = https://example.com/...
```

that would remove this issue completely, similar to how you configure endpoint url in s3 remotes. I remember we've discussed this during our 1on1, but I don't remember particular issues with it. Please feel free to remind me.

Reply:
Looking at it a bit more, it seems like that would be the correct approach, as webdavfs only cares about that base url. And it would also remove our old terrible hacks of converting webdav(s):// urls to http(s):// urls and getting confused about it in the code base 😄 But might be missing something.

Reply:
That solution to force our internal logic on the users feels wrong to me. That does the same thing as setting `fs_path` to root by default (i.e. without `url`). We can discuss changing config for sure, but that should be a separate discussion (not motivated by this bug at least).